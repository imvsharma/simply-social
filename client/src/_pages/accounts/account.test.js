/* 
    1. One component should have only one snapshot. - done
    2. Testing props.
    3. Testing data types.
    4. Event testing.
    5. Testing conditions.
    6. State's Testing.
*/
import React, { Component } from 'react';
import { shallow, mount } from 'enzyme';
import Account from './accounts';
import configureStore from 'redux-mock-store';

describe('Account component',() => {
    const mockStore = configureStore();
    let store, component;

    beforeEach(() => {
        store = mockStore();
        component = shallow(<Account store={store} />)
    })

    it('should render account component', () => {
        expect(component.exists()).toBe(true)
    });

    it('should render initial layout', () => {
        expect(component.getElements()).toMatchSnapshot();
    })
})
