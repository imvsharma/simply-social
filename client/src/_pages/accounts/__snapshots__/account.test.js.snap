// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component should render initial layout 1`] = `
Array [
  <Context.Provider
    value={null}
  >
    <Account
      dispatch={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    />
  </Context.Provider>,
]
`;
